addons/addon-stack-policy/packages/stack-policy/lib/steps/__test__/update-cfn-stack-policy.test.js

```diff
@@ -44,36 +47,60 @@ describe('UpgradeToUserId', () => {
       CloudFormation,
     };
     service = await container.find('UpdateCfnStackPolicy');
-    service._settings = {
-      get: settingName => {
-        if (settingName === 'enableEgressStore') {
-          return 'true';
-        }
-        if (settingName === 'backendStackName') {
-          return 'backendStackName';
-        }
-        return undefined;
-      },
-    };
+    settings = await container.find('settings');
+    settings.get = jest.fn(key => {
+      if (key === 'isAppStreamEnabled') {
+        return true;
+      }
+      if (key === 'enableEgressStore') {
+        return 'true';
+      }
+      if (key === 'backendStackName') {
+        return 'backendStackName';
+      }
+      return undefined;
+    });
   });
 
   describe('Run post deployment step', () => {
-    it('should successful update policy without stack policy', async () => {
+    it('should successfully update policy without stack policy', async () => {
```

Comment:
Perhaps rename to `should successfully update policy when old stack policy doesn't exist` for clarity